subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -522,6 +522,18 @@ static int dc_rx_data_handler(const struct nct_evt *nct_evt)
 		.topic = nct_evt->param.dc->topic,
 	};
 
+	if (nrf_cloud_detect_disconnection_request(nct_evt->param.dc->data.ptr)) {
+		LOG_DBG("nRF Cloud has requested us to disconnect, "
+			"likely due to device deletion");
+		int err = nrf_cloud_disconnect();
+
+		if (err < 0) {
+			LOG_ERR("nRF Cloud disconnection-on-request "
+				"failure, error: %d", err);
+		}
+		return 0;
```

Comment:
remove the return here so that `nfsm_set_current_state_and_notify` gets called.
i'm not sure if it matters, but it might be best, event-wise, to perform the disconnect after `nfsm_set_current_state_and_notify` is called?

Reply:
Sorry, I must have forgotten to add my changes to the most recent commit. Let me try again.

Reply:
Should be as you describe now